cardano_node_tests/utils/requirements.py

```diff
@@ -93,3 +95,42 @@ def collect_executed_req(base_dir: pl.Path) -> dict:
         id_collected["url"] = req_rec.get("url") or ""
 
     return collected
+
+
+def get_mapped_req(mapping: pl.Path, executed_req: dict) -> dict:
+    """Get mapped requirements."""
+    with open(mapping, encoding="utf-8") as in_fp:
+        requirements_mapping = json.load(in_fp)
+
+    for group, reqs in requirements_mapping.items():
+        for req_id, dependencies in reqs.items():
+            url = None
+            dependencies_success = []
+            dependencies_failures = []
+
+            for p_req in dependencies:
+                p_status = executed_req.get(group, {}).get(p_req, {}).get("status")
```

Comment:
Optional: you can declare variable like

```python
executed_group = executed_req.get(group) or {}
```

outside of the loop so you don't need to look it up over and over again.